middleware/sendJSON.js

```diff
@@ -1,71 +1,45 @@
-var check = require('check-types'),
-    es = require('elasticsearch'),
-    logger = require( 'pelias-logger' ).get( 'api' ),
-    exceptions = require('elasticsearch-exceptions/lib/exceptions/SupportedExceptions');
+const _ = require('lodash');
+const es = require('elasticsearch');
+const logger = require( 'pelias-logger' ).get( 'api' );
+const PeliasParameterError = require('../sanitizer/PeliasParameterError');
 
-// create a list of regular expressions to match against.
-// note: list created when the server starts up; for performance reasons.
-var exceptionRegexList = exceptions.map( function( exceptionName ){
-  return new RegExp( '^' + exceptionName );
-});
+function isTimeoutError(error) {
+  return error instanceof es.errors.RequestTimeout;
+}
+
+function isElasticsearchError(error) {
+  const knownErrors = [ es.errors.NoConnections,
+                        es.errors.ConnectionFault ];
+
+  return knownErrors.some(function(esError) {
+    return error instanceof esError;
+  });
+}
 
 function sendJSONResponse(req, res, next) {
 
   // do nothing if no result data set
-  if (!res || !check.object(res.body) || !check.object(res.body.geocoding)) {
+  const geocoding = _.get(res, 'body.geocoding');
+
+  if (!_.isObject(geocoding)) {
```

Comment:
`null` in JavaScript is an object, recently I've been using `_.isPlainObject(value)`.

what's your take?

Reply:
`_.isPlainObject` is probably a good idea in general.

In this case I'm not too concerned with checking the state of the `res.body.geocoding` object too carefully, as it is set to an object [unconditionally in middleware/geocodeJSON](https://github.com/pelias/api/blob/master/middleware/geocodeJSON.js#L41), but it's an easy change to make, so why not